istioctl/pkg/multicluster/remote_secret.go

```diff
@@ -1,588 +0,0 @@
-// Copyright Istio Authors.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package multicluster
-
-import (
-	"bytes"
-	"context"
-	"fmt"
-	"io"
-	"io/ioutil"
-	"strings"
-
-	"github.com/spf13/cobra"
-	"github.com/spf13/pflag"
-	v1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/serializer/json"
-	"k8s.io/apimachinery/pkg/runtime/serializer/versioning"
-	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-
-	//  to avoid 'No Auth Provider found for name "gcp"'
-	_ "k8s.io/client-go/plugin/pkg/client/auth"
-	"k8s.io/client-go/tools/clientcmd/api"
-	"k8s.io/client-go/tools/clientcmd/api/latest"
-
-	"istio.io/istio/operator/cmd/mesh"
-	"istio.io/istio/operator/pkg/helm"
-	"istio.io/istio/pkg/config/constants"
-	"istio.io/istio/pkg/config/labels"
-	"istio.io/istio/pkg/kube"
-	"istio.io/istio/pkg/kube/secretcontroller"
-)
-
-var (
-	codec  runtime.Codec
-	scheme *runtime.Scheme
-)
-
-func init() {
-	scheme = runtime.NewScheme()
-	utilruntime.Must(v1.AddToScheme(scheme))
-	opt := json.SerializerOptions{
-		Yaml:   true,
-		Pretty: false,
-		Strict: false,
-	}
-	yamlSerializer := json.NewSerializerWithOptions(json.DefaultMetaFactory, scheme, scheme, opt)
-	codec = versioning.NewDefaultingCodecForScheme(
-		scheme,
-		yamlSerializer,
-		yamlSerializer,
-		v1.SchemeGroupVersion,
-		runtime.InternalGroupVersioner,
-	)
-}
-
-const (
-	remoteSecretPrefix = "istio-remote-secret-"
-	configSecretName   = "istio-kubeconfig"
-	configSecretKey    = "config"
-)
-
-func remoteSecretNameFromClusterName(clusterName string) string {
-	return remoteSecretPrefix + clusterName
-}
-
-func clusterNameFromRemoteSecretName(name string) string {
-	return strings.TrimPrefix(name, remoteSecretPrefix)
-}
-
-// NewCreateRemoteSecretCommand creates a new command for joining two contexts
-// together in a multi-cluster mesh.
-func NewCreateRemoteSecretCommand() *cobra.Command {
```

Comment:
This is still used and essential

Reply:
probably I should first only remove "apply" "describe" and "generate" first for safety